### ISSUE TITLE
Video tagged with "This video contains mature or explicit content" not tagged when federated

### ISSUE THREAD

**Issue by gegeweb** [Chocobozzz/PeerTube#501]
* **PeerTube version or URL:** 1.0.0-beta3

Video tagged with "This video contains mature or explicit content" on the original instance aren't tagged on remote instances.

No examples, I've blacklisted the concerned videos coming from legal.p2p.tube on my instance, now I can't paste the UUIDs that are truncated.

**Comment by gegeweb** [comment]
I've temporarily unblocked this on my instance (peertube.gegeweb.eu): d31b929a-8d78-48d4-9647-8b71732a6514 where it is not tagged.
On the original instance peertube.p2p.legal it is tagged.

Don't know what looking on the logs.

**Comment by Chocobozzz** [comment]
Send me the log by email. Maybe the video was first not tagged NSFW, and then updated as NSFW and the update was not sent to your server (connection issue or whatever).

Add ability to hide NSFW content with an admin setting would be interesting, feel free to create another issue.

**Comment by gegeweb** [comment]
Only this that's not be a client request in the logs for this UUID :

```
[peertube.gegeweb.eu:443] 2018-04-06 05:45:31.263 info: Fetching remote video https://tube.p2p.legal/videos/watch/d31b929a-8d78-48d4-9647-8b71732a6514.
[peertube.gegeweb.eu:443] 2018-04-06 05:45:31.499 info: Remote video with uuid d31b929a-8d78-48d4-9647-8b71732a6514 inserted.
[peertube.gegeweb.eu:443] 2018-04-06 05:45:31.849 info: Adding likes of video d31b929a-8d78-48d4-9647-8b71732a6514.
[peertube.gegeweb.eu:443] 2018-04-06 05:45:31.850 info: Adding 0 like to video d31b929a-8d78-48d4-9647-8b71732a6514.
[peertube.gegeweb.eu:443] 2018-04-06 05:45:31.868 info: Adding dislikes of video d31b929a-8d78-48d4-9647-8b71732a6514.
[peertube.gegeweb.eu:443] 2018-04-06 05:45:31.869 info: Adding 0 dislike to video d31b929a-8d78-48d4-9647-8b71732a6514.
[peertube.gegeweb.eu:443] 2018-04-06 05:45:31.886 info: Adding shares of video d31b929a-8d78-48d4-9647-8b71732a6514.
[peertube.gegeweb.eu:443] 2018-04-06 05:45:32.103 info: Adding comments of video d31b929a-8d78-48d4-9647-8b71732a6514.
```
That is th output of node.
Are there any other more verbose logs?

**Comment by gegeweb** [comment]
This recent video, I think It's tagged on upload, but NSFW not federated:

![](https://framapic.org/TKY5eBkTZbwB/PTDRPc7HQXAr)

![](https://framapic.org/IeOmsGNpLwom/lWwfy4ihkOYa)
(feel free to remove the picture)

**Logs**
```
# cat /var/log/peertube/peertube.log | grep c530c70b-c631-402b-96f6-c0177b33d89b
[peertube.gegeweb.eu:443] 2018-04-14 01:30:06.171 info: Fetching remote video https://video.tedomum.net/videos/watch/c530c70b-c631-402b-96f6-c0177b33d89b.
[peertube.gegeweb.eu:443] 2018-04-14 01:30:06.614 info: Remote video with uuid c530c70b-c631-402b-96f6-c0177b33d89b inserted.
[peertube.gegeweb.eu:443] 2018-04-14 01:30:06.664 info: Adding likes of video c530c70b-c631-402b-96f6-c0177b33d89b.
[peertube.gegeweb.eu:443] 2018-04-14 01:30:06.665 info: Adding 0 like to video c530c70b-c631-402b-96f6-c0177b33d89b.
[peertube.gegeweb.eu:443] 2018-04-14 01:30:06.682 info: Adding dislikes of video c530c70b-c631-402b-96f6-c0177b33d89b.
[peertube.gegeweb.eu:443] 2018-04-14 01:30:06.683 info: Adding 0 dislike to video c530c70b-c631-402b-96f6-c0177b33d89b.
[peertube.gegeweb.eu:443] 2018-04-14 01:30:06.699 info: Adding shares of video c530c70b-c631-402b-96f6-c0177b33d89b.
[peertube.gegeweb.eu:443] 2018-04-14 01:30:07.234 info: Adding comments of video c530c70b-c631-402b-96f6-c0177b33d89b.
```
Nothing else at this time about this UUID on the logs.
I'll block the video. Don't want to have untagged porn available from my instance! ;)

**Comment by gegeweb** [comment]
Sorry for the inconvenience! ;)
It's not a bug, it comes from the user's configuration.
Display NSFW was enabled.
That said, I didn't quite understand the subtlety of this option.

Thanks to @Chocobozzz for the private help on Mastodon.

I've disabled this setting and everithing works fine.
